File: app/src/main/java/com/airsaid/example/custom/processor/AlertDialogActionProcessor.java
package com.airsaid.example.custom.processor;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.airsaid.sample.core.exception.SampleFailedException;
import com.airsaid.sample.core.processor.SampleInterfaceProcessor;
import com.airsaid.sample.api.Extension;

/**
 * @author JackChen
 */
@Extension
public class AlertDialogActionProcessor extends SampleInterfaceProcessor<AlertDialog> {
  @Override
  public void execute(@NonNull final AppCompatActivity context, final AlertDialog dialog)
      throws SampleFailedException {
    dialog.show();
  }
}

File: app/src/main/java/com/airsaid/example/custom/processor/AlertDialogActionProcessor.kt
package com.airsaid.example.custom.processor

import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.airsaid.sample.api.Extension
import com.airsaid.sample.core.processor.SampleInterfaceProcessor

/**
 * Demonstrates how to customize the corresponding action to be
 * performed when the sample item is clicked.
 *
 * @author airsaid
 */
@Extension
class AlertDialogActionProcessor : SampleInterfaceProcessor<AlertDialog>() {
  override fun execute(context: AppCompatActivity, obj: AlertDialog) {
    obj.show()
  }
}
